### PR TITLE
Feature/profile page

### PR DESCRIPTION
**Pull request by herve-perel**
## Description

Please include a summary of the change or which issue is fixed.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [x] Frontend
- [ ] Backend
- [ ] Other: .....

## General checklist:

- [x] My code follows the style guidelines of this project
- [x] I ran `bun run format`/`bun run lint` before commit
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation where needed
- [ ] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] I have updated `develop` and merged to my branch before submitting pull request
- [x] My pull request generate no conflicts with `develop` branch
- [x] I requested code review from other team members

## Frontend checklist:

- [x] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [x] My `Javascript` generate no new console errors
- [x] I tested my code cross browsers
- [x] My slice is pixel perfect for both desktop and mobile according to design
- [ ] I conducted basic QA to assure all features are working
- [x] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [ ] I followed guidelines for naming `Typescript` variables
- [ ] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by JorisPineau — DISMISSED** [review]

**Review by Abecker37 — DISMISSED** [review]

**Review by JorisPineau — APPROVED** [review]

**Review by Abecker37 — DISMISSED** [review]

File: packages/frontend/web/public/images/placeholder.jpg
Comment (YDrogen) [inline packages/frontend/web/public/images/placeholder.jpg:1]: placeholder de ?

File: packages/frontend/web/src/components/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        className={`
+        rounded-xl bg-[#A0B6C5] p-6 shadow transition-all sm:w-[50%] lg:w-[35%]
+        ${openModal ? 'scale-100 opacity-100' : 'scale-125 opacity-0'}
+      `}
+      >
+        <button
+          onClick={onClose}
+          className='absolute right-2 top-2 h-10 w-10 rounded-lg bg-white p-1 text-gray-400 hover:bg-gray-50 hover:text-gray-600'
+        >
+          <img src='./icons/cross.svg' />
```
Comment (YDrogen) [inline packages/frontend/web/src/components/Modal.tsx:31]: Mauvais chemin

File: packages/frontend/web/src/components/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      onClick={onClose}
+      className={`
+      fixed inset-0 flex items-center justify-center transition-colors
+      ${openModal ? 'visible bg-black/50' : 'invisible'}
+    `}
+    >
+      <div
+        onClick={(event) => {
+          event.stopPropagation();
+        }}
```
Comment (YDrogen) [inline packages/frontend/web/src/components/Modal.tsx:21]: Pourquoi ?

File: packages/types/src/profile.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+export type MusiqueButtonStates = {
```
Comment (YDrogen) [inline packages/types/src/profile.ts:1]: Pas mal le franglais

File: packages/types/src/profile.ts
Comment (YDrogen) [inline packages/types/src/profile.ts:1]: Pourquoi mettre les types ici ? Ils seront utilisés dans le backend aussi ?

File: packages/types/src/modal.ts
Comment (YDrogen) [inline packages/types/src/modal.ts:1]: Pourquoi mettre les types ici ? Ils seront utilisés dans le backend aussi ?

File: packages/frontend/web/src/pages/MyPreferences.tsx [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+            languagesButtonStates['french']
+              ? 'text-primary bg-white'
+              : 'bg-gray-400 text-black'
+          }`}
+        >
+          {'French'}
+        </button>
+      </div>
+      <div className='mt-5 flex justify-between'>
+        <p className='ms-[5%] text-xl'>{'accept smocking'}</p>
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/MyPreferences.tsx:170]: "smock" signifie "blouse"

File: packages/frontend/web/src/pages/MyPreferences.tsx [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+            handleLanguageClick('french');
+          }}
+          className={`h-8 w-20 rounded-full ${
+            languagesButtonStates['french']
+              ? 'text-primary bg-white'
+              : 'bg-gray-400 text-black'
+          }`}
+        >
+          {'French'}
+        </button>
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/MyPreferences.tsx:167]: Code dupliqué, tu peux faire une boucle

File: packages/frontend/web/src/pages/MyPreferences.tsx [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+      </Link>
+      <h1 className='text-bold ms-[5%] text-2xl'>{'My profil preferences'}</h1>
+      <div className=' ms-[5%] mt-5'>
+        <textarea
+          placeholder='Presentation'
+          className=' h-28 w-[95%] rounded-lg p-2 text-black'
+        />
+      </div>
+      <p className='ms-[5%] mt-5'>{'Music'}</p>
+      <div className='mt-2 flex justify-around'>
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/MyPreferences.tsx:56]: idem

File: packages/frontend/web/src/pages/MyPreferences.tsx [hunk 7]
```diff
@@ -0,0 +1,206 @@
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
+
+import type {
+  LanguagesButtonStates,
+  MusiqueButtonStates,
+} from '@app/types/src/profile';
+
+export default function MyPreferences() {
+  const [musiqueButtonStates, setMusiqueButtonStates] = useState({
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/MyPreferences.tsx:10]: Tu peux en profiter pour typer le useState

File: packages/frontend/web/src/router.tsx [hunk 8]
```diff
@@ -13,6 +17,22 @@ const router = createBrowserRouter([
         path: '/',
         element: <Home />,
       },
+      {
+        path: '/profil',
```
Comment (YDrogen) [inline packages/frontend/web/src/router.tsx:21]: Lien en français ?

**Review by YDrogen — CHANGES_REQUESTED** [review]
Étrange
image

**Review by JorisPineau — APPROVED** [review]

**Review by Abecker37 — APPROVED** [review]

**Review by JorisPineau — APPROVED** [review]

File: packages/frontend/web/src/pages/Profile.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+      <Link to='/my-opinions'>
+        <div className=' mt-5 flex justify-between'>
+          <p className='ms-[5%]'>{'My opinions'}</p>
+          <img src='/icons/right-arrow.svg' className='me-[5%]' />
+        </div>
+      </Link>
+      <div className='mx-auto w-[90%] border-b border-white' />
+      <Link to='/my-preferences'>
+        <div className=' mt-5 flex justify-between'>
+          <p className='ms-[5%]'>{'My profil preferences'}</p>
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/Profile.tsx:43]:
```suggestion
          <p className='ms-[5%]'>{'My profile preferences'}</p>
```

File: packages/frontend/web/src/pages/Profile.tsx [hunk 10]
```diff
@@ -0,0 +1,56 @@
+import { Link } from 'react-router-dom';
+
+export default function Profile() {
+  return (
+    <div className='mx-auto w-[85%] from-[#FFFFFF]/10 to-[#FFFFFF]/0 md:mt-28 md:h-[50rem] md:w-[40%] md:rounded-[1.5rem] md:bg-gradient-to-br md:p-5 md:shadow-2xl lg:ms-auto lg:w-[35rem] lg:py-5'>
+      <img
+        src='/images/user-placeholder.jpg'
+        alt='image-profil'
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/Profile.tsx:8]: profil : français
profile : anglais

File: packages/frontend/web/src/pages/MyPreferences.tsx [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+            ) as (keyof LanguagesButtonStates)[],
+          )}
+        </div>
+        <div className='mt-5 flex justify-between'>
+          <p className='ms-[5%] text-xl'>{'accept smoking'}</p>
+          <input
+            type='checkbox'
+            className='peer absolute left-1/2 h-5 w-full -translate-x-1/2 appearance-none rounded-md'
+          />
+          <span className='peer-checked:after:bg-primary me-[5%] ml-4 flex h-5 w-10 flex-shrink-0 items-center rounded-full bg-white  duration-300 ease-in-out after:h-5 after:w-5 after:rounded-full after:bg-black after:shadow-md after:duration-300  peer-checked:bg-white peer-checked:after:translate-x-5' />
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/MyPreferences.tsx:128]: double espaces

File: packages/frontend/web/src/pages/MyPreferences.tsx [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+      className={`h-8 w-20 rounded-full ${
+        buttonStates[buttonId]
+          ? 'text-primary bg-white'
+          : 'bg-gray-400 text-black'
+      }`}
+    >
+      {buttonId}
+    </button>
+  ));
+};
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/MyPreferences.tsx:51]: Pourquoi mettre ça dans des fonctions ?

**Review by JorisPineau — APPROVED** [review]

**Review by jaalazard — APPROVED** [review]

**Review by Abecker37 — APPROVED** [review]

**Review by YDrogen — APPROVED** [review]

**Pull request merged**